!!! TASK: Require PHP 8.2

robertlemke opened the pull request:
The minimum requirement for the Flow Framework, including all packages of its distribution, was raised to PHP 8.2.

robertlemke committed:
!!! TASK: Require PHP 8.2 or higher for Flow

The minimum requirement for the Flow Framework, including all packages of its distribution, was raised to PHP 8.2.

fcool commented:
What about the split repositories like "Utility\MediaTypes" "Utility\Files" and so on?

robertlemke commented:
> What about the split repositories like "Utility\MediaTypes" "Utility\Files" and so on?

I currently don't plan to change anything there which would require PHP 8.2, so we could leave them as they are, in case the packages are used elsewhere without Flow. On the other hand, that would increase the complexity for maintaining this repository, so maybe bump PHP to 8.2 for all of them, too?

fcool commented:
I'm sure we will mess up, if we would use different minimal requirements. So I would probably go for changing them all.

If we were going to maintain them differently, we probably needed to set up tests for all of them in all PHP variants, as we are not going to test them together with the main repo any more.

So to save us work, from which probably next to no one would benefit, my opionion is: Let's keep it simple.

And as we are going to raise all their version numbers to 9.0.0 it is absolutely fine to introduce higher dependencies for new Majors.

robertlemke commented:
@fcool Makes sense – I force-pushed a new commit which raises the PHP requirement for all packages of the distribution.

I'm curious what Psalm will say this time and what to do about potential errors …

robertlemke commented:
I need to adjust dependencies and code related to TYPO3 Fluid for this to work.

robertlemke committed:
TASK: Require PHP 8.2 compatible Fluid version

The Flow Development Collection now depends on Fluid 2.8 or higher in order to provide compatibility with PHP 8.2.

robertlemke committed:
TASK: Require Doctrine ORM 2.12.0 or higher

Flow now requires at least Doctrine ORM 2.12, since using an earlier version with PHP 8.2 resulted in various broken functional tests in the Flow development collection.

kdambekalns reviewed (commented):
Post-merge 🎉 from me!
